src/Subscriptions/SubscriptionPhase.php

```diff
@@ -463,28 +471,57 @@ private function add_interval( $date, $times = 1 ) {
 	}
 
 	/**
-	 * Get next period.
-	 *
+	 * Substract the interval of this subscription phase from the specified date.
+	 * 
+	 * @param DateTimeImmutable $date  Date to substract interval period from.
+	 * @param int               $times Number of times to substract interval.
+	 * @return DateTimeImmutable
+	 */
+	public function substract_interval( $date, $times = 1 ) {
+		// If times is zero there is nothing to add.
+		if ( 0 === $times ) {
+			return $date;
+		}
+
+		// Multiply date interval.
+		return $date->sub( $this->interval->multiply( $times ) );
+	}
+
+	/**
+	 * Get period for the specified start date.
+	 * 
+	 * @param DateTimeInterface $start_date Start date.
 	 * @return SubscriptionPeriod|null
 	 */
-	public function get_next_period() {
-		if ( $this->all_periods_created() ) {
+	public function get_period( DateTimeInterface $start_date = null ) {
+		if ( null === $start_date ) {
 			return null;
 		}
 
-		$start = $this->get_next_date();
-
-		if ( null === $start ) {
+		if ( $this->start_date > $start_date ) {
 			return null;
 		}
 
-		$end = $this->add_interval( $start );
+		$end_date = $this->add_interval( $start_date );
```

Comment:
```
Parameter #1 $date of method Pronamic\WordPress\Pay\Subscriptions\SubscriptionPhase::add_interval() expects Pronamic\WordPress\DateTime\DateTimeImmutable, Pronamic\WordPress\DateTime\DateTimeInterface given.
```

```suggestion
		$end_date = $this->add_interval( DateTimeImmutable::create_from_interface( $start_date ) );
```

Reply:
Resolved in https://github.com/pronamic/wp-pay-core/pull/129/commits/3bd2e275c479067d033192b9c686a53ad517c08a.